### PR TITLE
Add tests for MUL op + fix MUL op arithmetics to not panic on overflow

### PR DESCRIPTION
**Pull request by dezuman**

### CODE REVIEW

File: src/narmvm.rs [hunk 1]
```diff
@@ -319,7 +319,8 @@ impl NarmVM{
                 },
                 //0100_0011_01xx_xyyy MUL T1 flags
                 0b0100_0011_0100_0000 => {
-                    self.sreg[reg2] = valuen * valuem;
+                    let (result, _) = (valuen as u32).overflowing_mul(valuem as u32);
```
Comment (Earlz) [inline src/narmvm.rs:322]: To be fair, this wouldn't actually panic in release mode, but always a good practice to explicitly use `overflowing_XXX` when dealing with anything that can overflow/underflow. I had forgotten here

**Review by Earlz — APPROVED** [review]

File: src/narmvm.rs (hunk 1 above)
Comment (dezuman) [reply src/narmvm.rs:322]: Right, read a few days ago and then promptly forgot about the only-in-testing overflow checks. But best practice is best practice!
